include/mega65/shres.h

```diff
@@ -0,0 +1,123 @@
+/**
+ * @file shres.h
+ * @brief MEGA65 SYSPART Shared Resources Access API
+ *
+ * This API provides access to shared resources stored in the MEGA65 system
+ * partition (SYSPART), including files such as fonts, icons, or other binary
+ * assets.
+ *
+ * Resources are accessed via a special trap instruction and a sector-based
+ * directory format. This API allows enumeration, opening, reading, and seeking
+ * within those shared resources.
+ */
+
+#ifndef SHRES_H
+#define SHRES_H
+
+/// Triggers the low-level SYSPART shared resource trap.
+extern void shres_trap(void);
+
+/// Registers used for communicating with the shared resource trap.
+extern unsigned char shres_regs[5];
```

Comment:
Does this need to be a global variable? With llvm-mos, returning it from `shres_trap(void)` is usually better as the compiler has more optimization opportunities (return by registers, skip unused values etc.). I would also prefer `shres_trap()` to use inline asm with llvm-mos. I'm a little rusty but would be happy to look into this. It would be helpful with test file in `tests/` for that purpose.